solutions/kafka-long-term-logging/google_cloud_function_restore/main.py

```diff
@@ -0,0 +1,37 @@
+from uuid import uuid4
+from opensearchpy import OpenSearch
+from google.cloud import storage
+import json
+import os
+
+storage_client = storage.Client()
+
+os_client = OpenSearch(
+    hosts = [{'host': os.getenv("OS_HOST"), 'port': os.getenv("OS_PORT")}],
+    http_compress = True, # enables gzip compression for request bodies
+    http_auth = (os.getenv("OS_USER"), os.getenv("OS_PWD")),
+    use_ssl = True,
+    verify_certs = False,
+    ssl_assert_hostname = False,
+    ssl_show_warn = False,
+)
+
+def restore_logs(request):
```

Comment:
the whole function will take a while for large indexes. How many docs are we expecting per gzip file?

Reply:
I am not 100% sure but I thought we have limit of 500 set by a sink.